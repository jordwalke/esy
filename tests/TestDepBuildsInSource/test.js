const {createTestEnv} = require('../harness');

let sandbox = createTestEnv({
  sandbox: __dirname
});

test('env', () => {
  let res = sandbox.exec('../../.bin/esy');
  expect(res.status).toBe(0);
  expect(res.stdout.toString()).toMatchSnapshot();
});

test('forces build to happen in $cur__target_dir', () => {
  let res = sandbox.exec(`
  rm -rf _build _install _esy_store
  ../../.bin/esy build
  `);
  expect(res.status).toBe(0);
  expect(sandbox.readFile('node_modules', 'dep', 'OK')).toBe('OK');
  expect(sandbox.readFile('_esy_store', '_build', 'dep-1.0.0', 'OK')).toBe('OK!!!');
});
