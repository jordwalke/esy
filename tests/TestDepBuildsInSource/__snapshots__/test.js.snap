exports[`test env 1`] = `
"
# EsySandboxVariables 
export PATH=\"/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin\"
export SHELL=\"env -i /bin/bash --norc --noprofile\"
export esy__platform=\"platform\"
export esy__architecture=\"architecture\"
export esy__target_platform=\"platform\"
export esy__target_architecture=\"architecture\"
# [BuiltIns]
export esy__sandbox=\"$ESY__SANDBOX\"
export esy__store=\"$ESY__STORE\"
export esy__install_tree=\"$esy__sandbox/_install\"
export esy__build_tree=\"$esy__sandbox/_build\"
export cur__name=\"TestDepOutOfSourceBuildIsFalse\"
export cur__version=\"1.0.0\"
export cur__root=\"$esy__sandbox/\"
export cur__depends=\"dep\"
export cur__target_dir=\"$esy__sandbox/_build\"
export cur__install=\"$esy__sandbox/_install\"
export cur__bin=\"$cur__install/bin\"
export cur__sbin=\"$cur__install/sbin\"
export cur__lib=\"$cur__install/lib\"
export cur__man=\"$cur__install/man\"
export cur__doc=\"$cur__install/doc\"
export cur__stublibs=\"$cur__install/stublibs\"
export cur__toplevel=\"$cur__install/toplevel\"
export cur__share=\"$cur__install/share\"
export cur__etc=\"$cur__install/etc\"
# [Custom Variables]
export OCAMLFIND_CONF=\"$cur__target_dir/_esy/findlib.conf\"
export PATH=\"$esy__store/_install/dep-1.0.0/bin:$PATH\"
export MAN_PATH=\"$esy__store/_install/dep-1.0.0/man:$MAN_PATH\"

# dep@1.0.0 $esy__sandbox/node_modules/dep/package.json
# [BuiltIns]
export dep__name=\"dep\"
export dep__version=\"1.0.0\"
export dep__root=\"$esy__store/_build/dep-1.0.0\"
export dep__depends=\"\"
export dep__target_dir=\"$esy__store/_build/dep-1.0.0\"
export dep__install=\"$esy__store/_install/dep-1.0.0\"
export dep__bin=\"$dep__install/bin\"
export dep__sbin=\"$dep__install/sbin\"
export dep__lib=\"$dep__install/lib\"
export dep__man=\"$dep__install/man\"
export dep__doc=\"$dep__install/doc\"
export dep__stublibs=\"$dep__install/stublibs\"
export dep__toplevel=\"$dep__install/toplevel\"
export dep__share=\"$dep__install/share\"
export dep__etc=\"$dep__install/etc\"

# TestDepOutOfSourceBuildIsFalse@1.0.0 $esy__sandbox/package.json
# [BuiltIns]
export testdepoutofsourcebuildisfalse__name=\"TestDepOutOfSourceBuildIsFalse\"
export testdepoutofsourcebuildisfalse__version=\"1.0.0\"
export testdepoutofsourcebuildisfalse__root=\"$esy__sandbox/\"
export testdepoutofsourcebuildisfalse__depends=\"dep\"
export testdepoutofsourcebuildisfalse__target_dir=\"$esy__sandbox/_build\"
export testdepoutofsourcebuildisfalse__install=\"$esy__sandbox/_install\"
export testdepoutofsourcebuildisfalse__bin=\"$testdepoutofsourcebuildisfalse__install/bin\"
export testdepoutofsourcebuildisfalse__sbin=\"$testdepoutofsourcebuildisfalse__install/sbin\"
export testdepoutofsourcebuildisfalse__lib=\"$testdepoutofsourcebuildisfalse__install/lib\"
export testdepoutofsourcebuildisfalse__man=\"$testdepoutofsourcebuildisfalse__install/man\"
export testdepoutofsourcebuildisfalse__doc=\"$testdepoutofsourcebuildisfalse__install/doc\"
export testdepoutofsourcebuildisfalse__stublibs=\"$testdepoutofsourcebuildisfalse__install/stublibs\"
export testdepoutofsourcebuildisfalse__toplevel=\"$testdepoutofsourcebuildisfalse__install/toplevel\"
export testdepoutofsourcebuildisfalse__share=\"$testdepoutofsourcebuildisfalse__install/share\"
export testdepoutofsourcebuildisfalse__etc=\"$testdepoutofsourcebuildisfalse__install/etc\"
"
`;
